Kickstarter-iOS/Views/Controllers/LandingPageViewController.swift

```diff
@@ -0,0 +1,227 @@
+  private let cardsStackView: UIStackView = { UIStackView(frame: .zero) }()
+  private let cardViewsStackView: UIStackView = { UIStackView(frame: .zero) }()
+  private let ctaButton: UIButton = { UIButton(frame: .zero) }()
+  private let descriptionLabel: UILabel = { UILabel(frame: .zero) }()
+  private let labelsStackView: UIStackView = { UIStackView(frame: .zero) }()
+  private let logoImageView: UIImageView = { UIImageView(frame: .zero) }()
+  private let pageControl: UIPageControl = { UIPageControl(frame: .zero) }()
+  private let rootStackView: UIStackView = { UIStackView(frame: .zero) }()
+  private let scrollView: UIScrollView = {
+    UIScrollView(frame: .zero)
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  private let titleLabel: UILabel = { UILabel(frame: .zero) }()
+  private let viewModel: LandingPageViewModelType = LandingPageViewModel()
+
+  // MARK: - Life cycle
+
+  public override func viewDidLoad() {
+    super.viewDidLoad()
+
+    self.configureViews()
+    self.setupConstraints()
+    self.ctaButton.addTarget(
+      self,
+      action: #selector(LandingPageViewController.ctaBUttonTapped),
+      for: .touchUpInside
+    )
+
+    self.viewModel.inputs.viewDidLoad()
+  }
+
+  // MARK: - View Model
+
+  public override func bindViewModel() {
+    super.bindViewModel()
+
+    self.viewModel.outputs.landingPageCards
+      .observeForUI()
+      .observeValues { [weak self] cards in
+        self?.configureCards(with: cards)
+      }
+  }
+
+  // MARK: - Styles
+
+  public override func bindStyles() {
+    super.bindStyles()
+
+    _ = self.backgroundImageView
+      |> backgroundImageViewStyle
+
+    _ = self.cardsStackView
+      |> cardsStackViewStyle
+
+    _ = self.ctaButton
+      |> ctaButtonStyle
+
+    _ = self.descriptionLabel
+      |> descriptionLabelStyle
+
+    _ = self.labelsStackView
+      |> labelsStackViewStyle
+
+    _ = self.logoImageView
+      |> logoImageViewStyle
+
+    _ = self.rootStackView
+      |> rootStackViewStyle
+
+    _ = self.scrollView
+      |> scrollViewStyle
+      |> \.delegate .~ self
+
+    _ = self.titleLabel
+      |> titleLabelStyle
+  }
+
+  // MARK: - Configuration
+
+  private func configureViews() {
+    _ = (self.backgroundImageView, self.view)
+      |> ksr_addSubviewToParent()
+      |> ksr_constrainViewToEdgesInParent()
+
+    _ = ([self.logoImageView, self.titleLabel, self.descriptionLabel], self.labelsStackView)
+      |> ksr_addArrangedSubviewsToStackView()
+
+    _ = (self.cardViewsStackView, self.scrollView)
+      |> ksr_addSubviewToParent()
+      |> ksr_constrainViewToEdgesInParent()
+
+    _ = ([self.scrollView, self.pageControl], self.cardsStackView)
+      |> ksr_addArrangedSubviewsToStackView()
+
+    let spacer1 = UIView()
+    let spacer2 = UIView()
+    let spacer3 = UIView()
+
+    _ = ([
+      spacer1,
+      self.labelsStackView,
+      self.cardsStackView,
+      spacer2,
+      spacer3,
```

Comment:
Why the two spacers after each other? Are these just views that need to stretch?

Reply:
The distribution of views of this layout is a bit different of what we usually do in the app (extra spacing on top and bigger space between cards and the button). I tried configuring the views in different ways, but the result was never satisfactory. I'm open to retry if you have strong feelings about it.

Reply:
Oh I guess I'm just not sure what the two spacers adjacent to each other would do that a single spacer view can't do? How are the spacers receiving their height?

Reply:
Without the other spacer, the top view would be too low on iPhone 5s. The heights are being calculated by the stackView